### ISSUE TITLE
Not all non-disabled extensions in `vk.xml` provide a `type`.

### ISSUE THREAD

**Issue by NicolBolas** [KhronosGroup/Vulkan-Docs#354]
The 1.0.25 gives `extension` elements in `vk.xml` a `type` attribute that describes whether it is a `device` or `instance` extension. But many newer non-"disabled" extensions don't provide this.

For example, VK_ARB_draw_indirect_count is [supposed to be a device extension](https://github.com/KhronosGroup/Vulkan-Docs/blob/1.0-VK_AMD_draw_indirect_count/doc/specs/vulkan/appendices/VK_AMD_draw_indirect_count.txt), but that is not in `vk.xml`. Several of NVIDIA's new extensions like VK_NV_external_memory_capabilities lack this as well.

This also might be a good place to employ RelaxNG to verify such things in the future. For example, the current `registry.rnc` content model for `extension` says this:

```
attribute type { text } ?,
attribute supported { StringGroup } ? ,
```

If you change it to this:

```
( attribute supported { "disabled" } | (
    attribute supported { StringGroup },
    attribute type { "device" | "instance" }
))
```

Then you could just run any RelaxNG validator on `vk.xml` and it would give you an error when someone forgets to provide a proper `type` attribute for a non-"disabled" extension.

**Comment by oddhack** [comment]
I've tagged the ones I knew about as of a few days ago. A couple of the extension appendices don't provide this information yet:


* @TobiasHector - for IMG_format_pvrtc. I know there's some lingering IP issue about including that appendix in the spec you have to sort out, but since it *is* tagged in the XML, maybe we could at least finish that tagging while the lawyers go on.

* @courtney-g - for EXT_debug_report. That's the one that needs a bunch of changes done to match the spec body / appendix structure of other extensions, too - the "extension type" goes into the appendix header that currently doesn't exist :-)

* I already dealt with VK_EXT_validation_flags internally, it will go out next week.

**Comment by TobiasHector** [comment]
Just going through my bugs - looks like these all got fixed at some point? Closing since AFAICT all extensions now have the correct type tagging.

**Comment by oddhack** [comment]
We did AFAIK fix them all. Enhancing the RNG schema is a good idea but slightly more complicated because of all the "placeholder" \<extension> tags (e.g. name="VK_NV_extension_101") which don't have assigned types yet. AFAICT RNG supports constraints like

    (no type and supported="disabled") or (valid type and supported="vulkan")

which should do, though (not intended to be actual RNC syntax). I'll track this as a background thing internally.

**Comment by andens** [comment]
@TobiasHector It has popped up again with `VK_AMD_mixed_attachment_samples`. Generally, what's the intended semantics when type is omitted from non-disabled extensions? The registry documentation says that type is optional, and if present it must be either 'device' or 'instance', but what does it mean if it's omitted? Can one assume it's 'device'?

**Comment by oddhack** [comment]
@andens all you should assume is that the extension author forgot to include this attribute in their MR.

**Comment by oddhack** [comment]
The AMD extension should be addressed in the 1.0.60 spec update. I'm looking at schema updates but it's a deep background issue.
